### PR TITLE
add 'what-is-an-object'

### PR DESCRIPTION
**Pull request by pavankalyan-codes**
### Description

Adding "What is Object" entry for General.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

**Comment by sonnynomnom** [comment]
This one requires two reviews from the three of us @Dusch4593 @yangc95.

File: content/general/concepts/what-is-an-object/what-is-an-object.md
Comment (sonnynomnom) [inline content/general/concepts/what-is-an-object/what-is-an-object.md:3]: Let's shorten the Description in the metadata:

```
Description: 'An object is an instance of a class. It has the properties and behaviors of its class.'
```

File: content/general/concepts/what-is-an-object/what-is-an-object.md
Comment (sonnynomnom) [inline content/general/concepts/what-is-an-object/what-is-an-object.md:22]: Let's only have 2 items here. So let's do:

```
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
  ```

File: content/general/concepts/what-is-an-object/what-is-an-object.md
Comment (sonnynomnom) [inline content/general/concepts/what-is-an-object/what-is-an-object.md:28]: lowercase object:

An object is an instance of a class. It is created when we instantiate the class. 

An object contains the properties and functionalities of its class. When an object is created, memory will be allocated for it.

File: content/general/concepts/what-is-an-object/what-is-an-object.md
Comment (sonnynomnom) [inline content/general/concepts/what-is-an-object/what-is-an-object.md:33]: Use triple tick java

```java
// Creating the Employee class
class Employee {
  int id;
  String firstName;
  String lastName;
  char middleInitial;
  float years;
}

// Creating five different objects from the Employee class
Employee tina = new Employee();
Employee louise = new Employee();
Employee linda = new Employee();
Employee bob = new Employee();
Employee gene = new Employee();
```

File: content/general/concepts/what-is-an-object/what-is-an-object.md
Comment (sonnynomnom) [inline content/general/concepts/what-is-an-object/what-is-an-object.md:45]: In the above example, we created five different objects from the class `Employee`.

**Review by sonnynomnom — CHANGES_REQUESTED** [review]
Made some changes. Please make the changes before it goes through review #2 from another maintainer.

Also, I wonder if you can add some history about Objects. When and why it was invented. And how it's used in programs today (OOP).

**Comment by yangc95** [comment]
Great first entry; make sure to stick to the [PR template](https://github.com/Codecademy/docs/blob/main/.github/PULL_REQUEST_TEMPLATE.md) when submitting a PR!

Looking forward to reviewing your entry ^^

**Review by yangc95 — APPROVED** [review]

**Comment by sonnynomnom** [comment]
Hi @pavankalyan-codes, make sure to go through these checkboxes again:

Screen Shot 2021-10-06 at 3 26 54 AM

**Review by sonnynomnom — APPROVED** [review]

**Comment by pavankalyan-codes** [comment]
> Congrats @pavankalyan-codes! Just merged and live:
> 
> https://www.codecademy.com/resources/docs/general/what-is-an-object
> 
> We might need to revisit this again to add some historical context for objects and OOP.

Thanks @sonnynomnom :)  Learnt a lot while making this contribution. Will try to make a couple of more contributions.

**Pull request merged**
